### PR TITLE
Implemented code to make the open orders flow functional(#1wut90a)

### LINKED ISSUE #81: Implemente support to make the open orders flow functional
## What are the acceptance criteria?

- [ ] ...

## Can you complete this feature request by yourself?

- [x] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by ymaheshwari1**
closes #81 #83

### CODE REVIEW

File: src/components/PicklistSize.vue [hunk 1]
```diff
@@ -8,31 +8,11 @@
...
-          </ion-item>
-          <ion-item>
-            <ion-radio value="e" slot="start" />  
-            <ion-label>30 {{ $t("orders") }}</ion-label>
-            <ion-note slot="end">38 {{ $t("items") }}</ion-note>
+        <ion-radio-group v-model="size" @ionChange="setPicklistSize()">
+          <ion-item v-for="count in preparePicklistSize()" :key="count">
+            <ion-radio slot="start" :value="count * 5"/>
+            <ion-label>{{ (count * 5) >= openOrders.total ? openOrders.total : count * 5}} {{ $t('orders') }}</ion-label>
+            <ion-note slot="end">10 items</ion-note>
```
[the remark this answers is not in the document]
Comment (ymaheshwari1) [reply src/components/PicklistSize.vue:15]: Sure Sir will add a comment to it.

**Review by dt2patel — APPROVED** [review]

File: src/store/modules/order/OrderState.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export default interface OrderState {
+  open: {
```
Comment (adityasharma7) [inline src/store/modules/order/OrderState.ts:2]: We should query and list together as we will keep inprogress/completed list too

File: src/store/modules/order/actions.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import { hasError, showToast } from '@/utils'
+import { translate } from '@/i18n'
+import * as types from './mutation-types'
+import { prepareOrderQuery } from '@/utils/solrHelper'
+
+
+const actions: ActionTree<OrderState, RootState> = {
+
+  // get open orders
+  async fetchOpenOrders ({ commit, state }, payload = {}) {
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:15]:
```suggestion
  async findOpenOrders ({ commit, state }, payload = {}) {
```

File: src/store/modules/order/actions.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      params.filters['shipmentMethodTypeId'] = { value: state.selectedShipmentMethods, op: 'OR' }
+    }
+
+    const orderQueryPayload = prepareOrderQuery(params)
+
+    try {
+      resp = await OrderService.fetchOpenOrders(orderQueryPayload);
+      if (resp.status === 200 && resp.data.grouped.orderId.matches > 0 && !hasError(resp)) {
+        const total = resp.data.grouped.orderId.ngroups
+        commit(types.ORDER_OPEN_UPDATED, {list: resp.data.grouped.orderId.groups, total})
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:46]: initially prepare orders object, assign it values and commit at end

File: src/views/AssignPickerModal.vue [hunk 5]
```diff
@@ -71,14 +80,153 @@ export default defineComponent({
...
+          showToast(translate('Failed to create picklist for orders'))
+        }
+      } catch (err) {
+        console.error(err)
+        showToast(translate('Something went wrong'))
+      }
+
+      emitter.emit("dismissLoader")
+    },
+    async fetchPickers() {
```
Comment (adityasharma7) [inline src/views/AssignPickerModal.vue:167]:
```suggestion
    async findPickers() {
```

File: src/views/AssignPickerModal.vue [hunk 6]
```diff
@@ -71,14 +80,153 @@ export default defineComponent({
...
+    }
+  },
   methods: {
+    isPickerSelected(id) {
+      return this.selectedPickers.some((picker) => picker.id == id)
+    },
     closeModal() {
       modalController.dismiss({ dismissed: true });
     },
+    pickerChanged(id) {
```
Comment (adityasharma7) [inline src/views/AssignPickerModal.vue:103]:
```suggestion
    changePicker(id) {
```

File: src/views/AssignPickerModal.vue [hunk 7]
```diff
@@ -71,14 +80,153 @@ export default defineComponent({
...
+      } else {
+        this.selectedPickers.push(this.pickers.find((picker) => picker.id == id))
+      }
+    },
+    async searchPicker () {
+      this.pickers = []
+      this.fetchPickers()
+    },
+    async printPicklist () {
+      if (!this.selectedPickers.length) {
```
Comment (adityasharma7) [inline src/views/AssignPickerModal.vue:117]: We should handle by disabling the button

File: src/views/AssignPickerModal.vue [hunk 8]
```diff
@@ -71,14 +80,153 @@ export default defineComponent({
...
+        formData.append("itemStatusId_o_"+index, "PICKITEM_PENDING")
+        formData.append("shipGroupSeqId_o_"+index, item.shipGroupSeqId)
+        formData.append("orderId_o_"+index, item.orderId)
+        formData.append("orderItemSeqId_o_"+index, item.orderItemSeqId)
+        formData.append("productId_o_"+index, item.productId)
+        formData.append("quantity_o_"+index, item.itemQuantity)
+        formData.append("inventoryItemId_o_"+index, item.inventoryItemId)
+        formData.append("picked_o_"+index, item.itemQuantity)
+        formData.append("_rowSubmit_o_"+index, "Y")
+      })
```
Comment (adityasharma7) [inline src/views/AssignPickerModal.vue:146]: We could skip an iteration
```suggestion
     int index = 0;
      orders.map((order) => {
        order.doclist.docs.map((item) => {
          formData.append("facilityId_o_"+index, this.currentFacility.facilityId)
        formData.append("shipmentMethodTypeId_o_"+index, item.shipmentMethodTypeId)
        formData.append("itemStatusId_o_"+index, "PICKITEM_PENDING")
        formData.append("shipGroupSeqId_o_"+index, item.shipGroupSeqId)
        formData.append("orderId_o_"+index, item.orderId)
        formData.append("orderItemSeqId_o_"+index, item.orderItemSeqId)
        formData.append("productId_o_"+index, item.productId)
        formData.append("quantity_o_"+index, item.itemQuantity)
        formData.append("inventoryItemId_o_"+index, item.inventoryItemId)
        formData.append("picked_o_"+index, item.itemQuantity)
        formData.append("_rowSubmit_o_"+index, "Y")
        index++;
        } 
      });
```

File: src/views/AssignPickerModal.vue [hunk 9]
```diff
@@ -71,14 +80,153 @@ export default defineComponent({
...
+        noConditionFind: 'Y',
+        orderBy: "firstName ASC",
+        filterByDate: "Y",
+        distinct: "Y",
+        fieldList: ["firstName", "lastName", "partyId"]
+      }
+
+      try {
+        const resp = await UtilService.getAvailablePickers(payload);
+        if (resp.status === 200 && !hasError(resp) && resp.data.count > 0) {
```
Comment (adityasharma7) [inline src/views/AssignPickerModal.vue:207]: We should not show "Something went wrong" error when there are no pickers found. It should be empty state.

File: src/components/ViewSizeSelector.vue [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+      // Added this check, if the viewSize options only have one option to select then making 0th index value as view size, and if having empty view than making view size as 0
+      // This condition will become true only in case when the orders total is equal to or less than 5, as we are generating size options with a difference of 5
+      if(viewSizeOptions.length <= 1) {
+        viewSizeOptions.length == 1 ? this.setViewSize(viewSizeOptions[0]) : this.setViewSize(0)
+      } else {
+        // Checking that whether the current size exists in the options available, if exist than not changing the size otherwise setting default size.
+        // This scenario occurs when we have selected a shipping method having for example 3 orders, so we set the size to 3
+        // Now when we de-select the shipping method or select another shipping method than the orders totals will increase,
+        // and thus 3 as an option will not be available, thus checking for below condition
+        viewSizeOptions.includes(this.viewSize) || (viewSizeOptions.includes(+process.env.VUE_APP_VIEW_SIZE) ? this.setViewSize(+process.env.VUE_APP_VIEW_SIZE) : this.setViewSize(viewSizeOptions[0]))
```
Comment (adityasharma7) [inline src/components/ViewSizeSelector.vue:80]: I think we should set it with query while fetching the order, if not already. This way one of the check would be reduced

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/AssignPickerModal.vue (hunk 9 above)
Comment (ymaheshwari1) [reply src/views/AssignPickerModal.vue:207]: Sir, we are not displaying it on the UI, we are just adding the error in console for debugging, also we already have empty state for the same.
I'll update the message to be more informative.

File: src/store/modules/order/actions.ts [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+        this.dispatch('product/getProductInformation', { orders })
+      } else {
+        console.error('No orders found')
+      }
+    } catch (err) {
+      console.error('error', err)
+      showToast(translate('Something went wrong'))
+    }
+
+    dispatch('updateViewSize', { size: orders.length, page: 'open' })
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:59]: Why a separate action here, we could directly commit

File: src/views/AssignPickerModal.vue [hunk 12]
```diff
@@ -71,14 +80,146 @@ export default defineComponent({
...
+    }
+  },
   methods: {
+    isPickerSelected(id) {
+      return this.selectedPickers.some((picker) => picker.id == id)
+    },
     closeModal() {
       modalController.dismiss({ dismissed: true });
     },
+    changePicker(id) {
```
Comment (adityasharma7) [inline src/views/AssignPickerModal.vue:103]:
```suggestion
    selectPicker(id) {
```

File: src/views/AssignPickerModal.vue [hunk 13]
```diff
@@ -71,14 +80,146 @@ export default defineComponent({
...
+      const picker = this.selectedPickers.some((picker) => picker.id == id)
+      if (picker) {
+        // if picker is already selected then removing that picker from the list on click
+        this.selectedPickers = this.selectedPickers.filter((picker) => picker.id != id)
+      } else {
+        this.selectedPickers.push(this.pickers.find((picker) => picker.id == id))
+      }
+    },
+    async searchPicker () {
```
Comment (adityasharma7) [inline src/views/AssignPickerModal.vue:112]: Why 2 methods?

**Pull request merged**